api/src/database/index.ts

```diff
@@ -243,3 +247,21 @@ export async function validateDatabaseExtensions(): Promise<void> {
 		}
 	}
 }
+
+async function validateDatabaseCharset(database?: Knex): Promise<void> {
+	database = database ?? getDatabase();
+	if (getDatabaseClient(database) === 'mysql') {
+		const directusResult = await database('information_schema.columns')
+			.select(`collation_name as collation`)
+			.where({ table_name: 'directus_users', column_name: 'id' })
+			.first();
+		const databaseResult = await database.select(database.raw(`@@collation_database as collation`)).first();
+		if (directusResult?.collation === databaseResult.collation) {
+			logger.warn(
+				`Directus tables collation (${directusResult.collation}) don't match your database default collation (${databaseResult.collation}).`
+			);
+		}
+		await database.raw("SET CHARACTER SET 'UTF8MB4'");
```

Comment:
Is this going to be a problem? UTF8MB4 seems like a good default, but also feels like something that might be a tad too opinionated 🤔